### ISSUE TITLE
SparkSQL dialect templating/parsing errors found with WHERE and DISTRIBUTE BY

### ISSUE THREAD

**Issue by anna-azizian** [sqlfluff/sqlfluff#3484]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When I try to lint/fix a query like this, I get "templating/parsing errors found" error: 
```sql
WITH cte AS (
    SELECT *
    FROM source
    WHERE col1 = 0
    DISTRIBUTE BY col1
),

SELECT *
FROM cte
```
I don't get errors without the WHERE condition or DISTRIBUTE BY

### Expected Behaviour

Sqlfluff should be able to lint/fix this query without errors.

### Observed Behaviour

Templating/parsing errors found.

### How to reproduce

Parse the query
```sql
WITH cte AS (
    SELECT *
    FROM source
    WHERE col1 = 0
    DISTRIBUTE BY col1
),

SELECT *
FROM cte
```

### Dialect

SparkSQL

### Version

sqlfluff, version 1.0.0

### Configuration

```
[sqlfluff]
dialect = databricks
templater = dbt
sql_file_exts = .sql
rules = core
exclude_rules = L022

[sqlfluff:indentation]
indented_joins = false
indented_using_on = true
template_blocks_indent = true

[sqlfluff:templater]
unwrap_wrapped_queries = true

[sqlfluff:templater:dbt]
project_dir = analytics_dbt_databricks/
profiles_dir = analytics_dbt_databricks/profiles/
profile = analytics_dbt
target = dev

[sqlfluff:rules]
tab_space_size = 4
max_line_length = 140
indent_unit = space
comma_style = trailing

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L030]
extended_capitalisation_policy = lower
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
